### PR TITLE
Fix VTK 9.3 failures

### PR DESCRIPTION
**Pull request by banesullivan**
The following are a list of failures being captured by our CI that tests against VTK master. Tracking here for the VTK 9.3 release

cc @alesgenova

### Composite core API

Something is going awry with the copying -- I suspect this was broken upstream

- [x] FAILED tests/core/test_composite.py::test_multi_block_copy - assert 139993061330560 == 139993061330944
    - Tracked in https://gitlab.kitware.com/vtk/vtk/-/issues/19143

### Composite plotting

I'm seeing this warning from VTK that might be the cause of many of these:

```
DeprecationWarning('Call to deprecated class vtkCompositePolyDataMapper2. (Please use vtkCompositePolyDataMapper instead) -- Deprecated since version 9.3.0.')
```

- [x] FAILED tests/plotting/test_plotting.py::test_orthographic_slicer - RuntimeError: test_orthographic_slicer Exceeded image regression error of 500.0 with an image error equal to: 3940.07712418308
- [x] FAILED tests/plotting/test_plotting.py::test_plot_composite_poly_complex - Failed: DID NOT WARN. No warnings of type (<class 'numpy.exceptions.ComplexWarning'>,) were emitted.
- [x] FAILED tests/plotting/test_plotting.py::test_plot_composite_bool - RuntimeError: test_plot_composite_bool Exceeded image regression error of 500.0 with an image error equal to: 7256.444444444465
- [x] FAILED tests/plotting/test_plotting.py::test_plot_composite_poly_component_norm - RuntimeError: test_plot_composite_poly_component_norm Exceeded image regression error of 500.0 with an image error equal to: 7393....
- [x] FAILED tests/plotting/test_plotting.py::test_plot_composite_many_options - RuntimeError: test_plot_composite_many_options Exceeded image regression error of 500.0 with an image error equal to: 6737.6326797...
- [x] FAILED tests/plotting/test_plotting.py::test_plot_composite_poly_component_single - RuntimeError: test_plot_composite_poly_component_single Exceeded image regression error of 500.0 with an image error equal to: 472...
- [x] FAILED tests/plotting/test_plotting.py::test_plot_composite_categories - RuntimeError: test_plot_composite_categories Exceeded image regression error of 500.0 with an image error equal to: 3068.726797385632
- [x] FAILED tests/plotting/test_plotting.py::test_plot_composite_poly_component_nested_multiblock - RuntimeError: test_plot_composite_poly_component_nested_multiblock Exceeded image regression error of 500.0 with an image error eq...
- [x] FAILED tests/plotting/test_plotting.py::test_plot_composite_lookup_table - RuntimeError: test_plot_composite_lookup_table Exceeded image regression error of 500.0 with an image error equal to: 3105.7555555...
- [x] FAILED tests/plotting/test_plotting.py::test_plot_composite_poly_scalars_opacity - RuntimeError: test_plot_composite_poly_scalars_opacity Exceeded image regression error of 500.0 with an image error equal to: 4034...

### Cubemap

This appears to be an issue between OSMesa and VTK 9.3

- [ ] FAILED tests/plotting/test_plotting.py::test_plot_cubemap_alone - RuntimeError: test_plot_cubemap_alone Exceeded image regression error of 500.0 with an image error equal to: 517.9764705882385

### CODE REVIEW

**Comment by banesullivan** [comment]
The cube map failure is concerning. The test is taking **71 seconds** and consuming 103.9 MB with VTK 9.3 (previously **1.61 seconds** consuming 36.1 MB) and it has worse quality:

| VTK 9.2 | VTK 9.3 (`master`) |
|---|---|
| plot_cubemap_alone_9 2 |  plot_cubemap_alone_9 3 |

**Comment by banesullivan** [comment]
cc @RobPasMue, @AlejandroFernandezLuces, et al. Some of changes in VTK 9.3 and this PR may affect usage of composite plotting

**Comment by RobPasMue** [comment]
Thanks for letting us know @banesullivan!

File: pyvista/core/composite.py [hunk 1]
```diff
@@ -1024,6 +1024,20 @@ def copy(self, deep=True):
...
+    def shallow_copy(self, to_copy: _vtk.vtkMultiBlockDataSet) -> None:
+        """Shallow copy the given multiblock to this multiblock.
+
+        Parameters
+        ----------
+        to_copy : pyvista.MultiBlock or vtk.vtkMultiBlockDataSet
+            Data object to perform a shallow copy from.
+
+        """
+        if pyvista.vtk_version_info >= (9, 3):
```
Comment (banesullivan) [inline pyvista/core/composite.py:1036]:
```suggestion
        if pyvista.vtk_version_info >= (9, 3):  # pragma: no cover
```

**Comment by banesullivan** [comment]
@pyvista/developers, ready for review! VTK 9.3 is looming

File: pyvista/core/composite.py [hunk 2]
```diff
@@ -1024,6 +1024,20 @@ def copy(self, deep=True):
...
+        """Shallow copy the given multiblock to this multiblock.
+
+        Parameters
+        ----------
+        to_copy : pyvista.MultiBlock or vtk.vtkMultiBlockDataSet
+            Data object to perform a shallow copy from.
+
+        """
+        if pyvista.vtk_version_info >= (9, 3):  # pragma: no cover
+            self.CompositeShallowCopy(to_copy)
```
Comment (banesullivan) [inline pyvista/core/composite.py:1037]: FYI: https://gitlab.kitware.com/vtk/vtk/-/issues/19143

File: tests/plotting/test_plotting.py [hunk 3]
```diff
@@ -3591,6 +3591,7 @@ def test_plot_texture_flip_y(texture):
 
 @pytest.mark.needs_vtk_version(9, 2, 0)
 @pytest.mark.skipif(CI_WINDOWS, reason="Windows CI testing segfaults on pbr")
+@pytest.mark.skipif(pv.vtk_version_info >= (9, 3), reason="This is broken on VTK 9.3")
```
Comment (banesullivan) [inline tests/plotting/test_plotting.py:3594]: This isn't broken per-se but rather I think something between VTK 9.3 and the version of libosmesa shipped with it is causing chaos

**Comment by akaszynski** [comment]
I can confirm from local testing that `xvfb` and cubemap still works on VTK 9.3 and (at least on my machine) I get the same timing for the `download_cubemap_park` example (terrible, clocking in at 2 minutes, but that's what you get with software rather than hardware).

**Review by akaszynski — APPROVED** [review]
I'm actually surprised that we didn't have to change that much for 9.3 compatibility. Thanks for looking into this @banesullivan.

I can't expect VTK to maintain the same degree of backwards compatibility as PyVista, and I'm fine with us spending the time serving as a higher level abstraction layer that makes VTK easier to use. It would have taken me hours and hours of debugging and searching through docs to figure out that `CompositeShallowCopy` is the way to go.

File: tests/plotting/test_plotting.py [hunk 4]
```diff
@@ -1606,7 +1606,7 @@ def test_link_views_camera_set(sphere, verify_image_cache):
 
 
 def test_orthographic_slicer(uniform):
-    uniform.set_active_scalars('Spatial Cell Data')
+    uniform.set_active_scalars('Spatial Point Data')
```
Comment (tkoyama010) [inline tests/plotting/test_plotting.py:1609]: @banesullivan Could you explain the reason why we failed the test in Cell Data?

Comment (banesullivan) [reply tests/plotting/test_plotting.py]: This was an artifact of me figuring out the composite rendering differences. I reverted

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**